### PR TITLE
Add FreeCol

### PR DESCRIPTION
**Pull request by rrigby**
http://www.freecol.org/

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build org.freecol.FreeCol

**Comment by nedrichards** [comment]
Is upstream actually dead for this? If not it'd be good to submit a quick 'is anyone interested in helping here' post.

**Comment by barthalion** [comment]
Seems to be active: https://github.com/FreeCol/freecol

CI failed due to missing icon (it may be just misnamed.)

**Comment by nedrichards** [comment]
Oh yeah, that's much more active than an old sourcefourge site. Cool, we need a ticket on there asking if upstream are interested in being involved as well.

**Comment by rrigby** [comment]
@barthalion, any ideas on what to do about the icon since appstream doesn't appear to support xpm?

Does flatpak-builder have access to image tools that could be used to generate icons?

**Comment by barthalion** [comment]
You can build imagemagick as part of the manifest and clean it up after the build with `cleanup: ["*"]` like here: https://github.com/flathub/com.github.ojubaorg.Othman/blob/2c132f4d4c4fe120b26fd7f120cd3ea39fc6ca77/com.github.ojubaorg.Othman.json#L44

…and use it to convert the file.

**Comment by rrigby** [comment]
@barthalion is it sufficient to use a binary for image magick? As it won't be included in the final flatpak.

**Comment by rrigby** [comment]
bot, build org.freecol.FreeCol

File: org.freecol.FreeCol.json [hunk 1]
```diff
@@ -50,7 +91,12 @@
         "cp -r /run/build/freecol/packaging/common/. /app/bin",
         "install -Dm644 /run/build/freecol/packaging/common/freecol.appdata.xml /app/share/appdata/freecol.appdata.xml",
         "install -Dm644 /run/build/freecol/packaging/common/freecol.desktop /app/share/applications/freecol.desktop",
-        "install -Dm644 /run/build/freecol/packaging/common/freecol_64x64.xpm /app/share/pixmaps/freecol.xpm"
+        "convert /run/build/freecol/data/freecol.png -gravity center -background none -extent 96x96 freecol.png",
+        "install -Dm644 freecol.png ${FLATPAK_DEST}/share/icons/hicolor/96x96/apps/freecol.png",
+        "for i in 32 48 64; do
```
Comment (barthalion) [inline org.freecol.FreeCol.json]: We need 128x128 too to make it look good in the app stores.

**Comment by nedrichards** [comment]
bot, build org.freecol.FreeCol

**Comment by barthalion** [comment]
Does the upstream team have IRC channel? It would be nice to at least give someone write access.

**Comment by rrigby** [comment]
@barthalion I can't seem to locate an active one, maybe just give access to the maintainers of the git repo? 

https://github.com/FreeCol/freecol

**Comment by barthalion** [comment]
I guess they will come here if they're interested.

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/org.freecol.FreeCol

Thanks!

**Pull request closed without merging**